app/services/move_to_problem_queue.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class MoveToProblemQueue
+
+  def self.move!(order_detail)
+    new(order_detail).move!
+  end
+
+  def initialize(order_detail)
+    @order_detail = order_detail
+  end
+
+  def move!
+    @order_detail.complete!
```

Comment:
I kinda think that inside this class we should set `@order_detail.force_completion = true` and call `@order_detail.complete!`. This would ensure that `move!` can succeed, and the call sites of `MoveToProblemQueue` do not need to know about the internal implementation detail `force_completion` of a problem order—that’s encapsulated by this class.

Reply:
Word. Good call.